### PR TITLE
ci(release): add winget releaser's step

### LINKED ISSUE #2599: Release to Windows Package Manager Community Repository (and set CI to release automatically)
Previous Discussions #2043

Thanks to #2578, after the next release, we will be able to register a manifest to https://github.com/microsoft/winget-pkgs and install prqlc.exe via `winget install` command.

Once the initial release on the repo is complete, I believe the followings could be used to send the manifests at release time automatically.
https://github.com/vedantmgoyal2009/winget-releaser
https://github.com/microsoft/winget-create

### PR DESCRIPTION
**Pull request by eitsupi**
Close #2599

This PR is intended to automate winget release (automated create PRs to <https://github.com/microsoft/winget-pkgs>)

Ref: https://github.com/kubernetes/minikube/blob/44b1e9fd20044244153d7bd5d87ee78084eeec3d/.github/workflows/winget.yml

### CODE REVIEW

**Comment by eitsupi** [comment]
@max-sixty To do this, One user's token must be stored in `secrets.WINGET_TOKEN`[^1].
[^1]: <https://github.com/vedantmgoyal2009/winget-releaser#getting-started->

I think @prql-bot would be the best choice for this, but the bot may not be able to respond if the user is asked to respond, like microsoft/winget-pkgs#112690

- Can @prql-bot sign the CLA? (See https://github.com/microsoft/winget-pkgs/pull/112690#issuecomment-1645296134)
- If @prql-bot is mentioned, can any of the maintainers notice it?

There are some bots that seem to work as well as @dotnet-winget-bot, so it will probably work.......

**Comment by max-sixty** [comment]
>     * Can @prql-bot sign the CLA? (See [New version: Kubernetes.minikube version 1.31.1 microsoft/winget-pkgs#112690 (comment)](https://github.com/microsoft/winget-pkgs/pull/112690#issuecomment-1645296134))

I have the login details, so I can login temporarily and sign it! (Or happy to share them with you if you prefer)

>     * If @prql-bot is mentioned, can any of the maintainers notice it?

It's hooked up to my email, and I could tag you when I see something (I would probably have a 70% success rate, I could set up a forwarding rule if we wanted better than that...)

**Review by max-sixty — APPROVED** [review]
Looks good!

**Comment by eitsupi** [comment]
Ok, thanks!
Please set `secrets.WINGET_TOKEN`.

We can't test the behavior of this, so fingers crossed it will work correctly in the next release......

**Comment by max-sixty** [comment]
> Please set `secrets.WINGET_TOKEN`.

I can do more research but do you already happen to know where to get the token?

**Comment by eitsupi** [comment]
> I can do more research but do you already happen to know where to get the token?

Settings -> Developer Settings -> Personal access tokens -> Tokens (clasic)

**Comment by max-sixty** [comment]
> Settings -> Developer Settings -> Personal access tokens -> Tokens (clasic)

~Sorry, but I mean — where do I get the value to put here!~

Sorry! To confirm, we just want the GitHub token here? Or we need a special Winget token?

**Comment by eitsupi** [comment]
> Sorry! To confirm, we just want the GitHub token here? Or we need a special Winget token?

A token with `public_repo` scope.
Please check <https://github.com/vedantmgoyal2009/winget-releaser#getting-started-> for details.

**Comment by max-sixty** [comment]
> > Sorry! To confirm, we just want the GitHub token here? Or we need a special Winget token?
> 
> A token with `public_repo` scope. Please check https://github.com/vedantmgoyal2009/winget-releaser#getting-started- for details.

Thanks. Can we use `PRQL_BOT_GITHUB_TOKEN`? I just created that; we can use that for all @prql-bot uses...

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -69,6 +69,20 @@ jobs:
...
+    needs: build-prqlc
+    if: github.ref_type == 'tag' && !inputs.dry-run
+    steps:
+      - name: publish
+        uses: vedantmgoyal2009/winget-releaser@v2
+        with:
+          identifier: PRQL.prqlc
+          version: ${{ github.ref }}
+          installers-regex: '^prqlc-.*-windows-.*\.zip$'
+          token: ${{ secrets.WINGET_TOKEN }}
```
Comment (max-sixty) [inline .github/workflows/release.yaml]:
```suggestion
          token: ${{ secrets.PRQL_BOT_GITHUB_TOKEN }}
```

**Comment by eitsupi** [comment]
> Thanks. Can we use `PRQL_BOT_GITHUB_TOKEN`?

I cannot answer because I do not know the scope of it, but I think it could probably be used.

**Comment by max-sixty** [comment]
> I cannot answer because I do not know the scope of it, but I think it could probably be used.

Yes! Sorry for my poor communication this morning...

**Comment by eitsupi** [comment]
> Yes! Sorry for my poor communication this morning...

Never mind. Thanks for the settings up dates!

**Pull request merged**